public/apps/configuration/top-nav-menu.tsx

```diff
@@ -13,62 +13,57 @@
  *   permissions and limitations under the License.
  */
 
-import React from 'react';
+import React, { useMemo } from 'react';
 import { DataSourceSelectableConfig } from 'src/plugins/data_source_management/public';
 import { DataSourceOption } from 'src/plugins/data_source_management/public/components/data_source_menu/types';
 import { AppDependencies } from '../types';
+import { setDataSourceInUrl } from '../../utils/datasource-utils';
 
 export interface TopNavMenuProps extends AppDependencies {
   dataSourcePickerReadOnly: boolean;
   setDataSource: React.Dispatch<React.SetStateAction<DataSourceOption>>;
   selectedDataSource: DataSourceOption;
 }
 
-const compatibleVersion = new Set([
-  '2.1',
-  '2.2',
-  '2.3',
-  '2.4',
-  '2.5',
-  '2.6',
-  '2.7',
-  '2.8',
-  '2.9',
-  '2.10',
-  '2.11',
-  '2.12',
-]);
+export const SecurityPluginTopNavMenu = React.memo(
```

Comment:
are we using memo to prevent re-populating the datasource picker on any component re-load?

https://react.dev/reference/react/memo

Reply:
Yes. without usememo, the callback function setting the datasource is causing re-render of this component, which is in turn setting the datasource again, causing an infinite loop. This is necessary after changes in core that would re-run the callback function even on initial render.

Reply:
oh..nice catch!